src/main/java/org/wise/portal/service/peergrouping/logic/impl/PeerGroupLogicServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package org.wise.portal.service.peergrouping.logic.impl;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.wise.portal.dao.peergroup.PeerGroupDao;
+import org.wise.portal.dao.run.RunDao;
+import org.wise.portal.domain.peergroup.PeerGroup;
+import org.wise.portal.domain.peergroup.impl.PeerGroupImpl;
+import org.wise.portal.domain.peergrouping.PeerGrouping;
+import org.wise.portal.domain.run.Run;
+import org.wise.portal.domain.workgroup.Workgroup;
+import org.wise.portal.service.peergroup.PeerGroupCreationException;
+import org.wise.portal.service.peergrouping.logic.PeerGroupLogicService;
+
+public abstract class PeerGroupLogicServiceImpl implements PeerGroupLogicService {
+
+  @Autowired
+  PeerGroupDao<PeerGroup> peerGroupDao;
+
+  @Autowired
+  private RunDao<Run> runDao;
+
+  public PeerGroup createPeerGroup(Workgroup workgroup, PeerGrouping peerGrouping)
+      throws PeerGroupCreationException {
+    List<Workgroup> workgroupsInPeriod = getWorkgroupsInSamePeriod(workgroup);
+    List<Workgroup> workgroupsInPeerGroup = peerGroupDao.getWorkgroupsInPeerGroup(peerGrouping,
+        workgroup.getPeriod());
```

Comment:
Maybe rename workgroupsInPeerGroup to workgroupsInAPeerGroup throughout the file because it sounds like the list is the workgroups that are in the peer group we are trying to create but it's actually the workgroups that are already in a peer group?

Likewise, maybe change workgroupsNotInPeerGroup to workgroupsNotInAPeerGroup?